### ISSUE TITLE
Nmap as root user

### ISSUE THREAD

**Issue by FernandoDoming** [kost/nmap-android#4]
I'm using this nmap version for Android arm64 and it works as expected for unprivileged users (`--unprivileged`) but reports everything as down / closed if executed as the root user.

Example here (Everything run as the root user):

```
/nmap --version                                                               <

Nmap version 6.49BETA4 ( https://nmap.org )
Platform: arm-unknown-linux-androideabi
Compiled with: liblua-5.2.3 openssl-1.0.2d nmap-libpcre-7.6 nmap-libpcap-1.7.3 nmap-libdnet-1.12 ipv6
Compiled without:
Available nsock engines: epoll poll select
```

```
/nmap -sP 10.0.30.0/24                                                        <

Starting Nmap 6.49BETA4 ( https://nmap.org ) at 2016-01-13 09:40 CET
mass_dns: warning: Unable to open /etc/resolv.conf. Try using --system-dns or specify valid servers with --dns-servers
mass_dns: warning: Unable to determine any DNS servers. Reverse DNS is disabled. Try using --system-dns or specify valid servers with --dns-servers
Nmap scan report for 10.0.30.170
Host is up.
Nmap done: 256 IP addresses (1 host up) scanned in 11.03 seconds

******************************************************************************
/nmap -sP 10.0.30.0/24 --unprivileged                                         <

Starting Nmap 6.49BETA4 ( https://nmap.org ) at 2016-01-13 09:40 CET
mass_dns: warning: Unable to open /etc/resolv.conf. Try using --system-dns or specify valid servers with --dns-servers
mass_dns: warning: Unable to determine any DNS servers. Reverse DNS is disabled. Try using --system-dns or specify valid servers with --dns-servers
Nmap scan report for 10.0.30.1
Host is up (0.0093s latency).
Nmap scan report for 10.0.30.2
Host is up (0.013s latency).
Nmap scan report for 10.0.30.29
Host is up (0.019s latency).
Nmap scan report for 10.0.30.39
Host is up (0.0078s latency).
Nmap scan report for 10.0.30.46
Host is up (0.015s latency).
Nmap scan report for 10.0.30.52
Host is up (0.072s latency).
Nmap scan report for 10.0.30.59
Host is up (0.050s latency).
Nmap scan report for 10.0.30.67
Host is up (0.013s latency).
Nmap scan report for 10.0.30.79
Host is up (0.088s latency).
Nmap scan report for 10.0.30.81
Host is up (0.010s latency).
Nmap scan report for 10.0.30.90
Host is up (0.039s latency).
Nmap scan report for 10.0.30.91
Host is up (0.016s latency).
Nmap scan report for 10.0.30.125
Host is up (0.057s latency).
Nmap scan report for 10.0.30.126
Host is up (0.076s latency).
Nmap scan report for 10.0.30.127
Host is up (0.058s latency).
Nmap scan report for 10.0.30.130
Host is up (0.0096s latency).
Nmap scan report for 10.0.30.132
Host is up (0.036s latency).
Nmap scan report for 10.0.30.134
Host is up (0.031s latency).
Nmap scan report for 10.0.30.156
Host is up (0.019s latency).
Nmap scan report for 10.0.30.164
Host is up (0.0055s latency).
Nmap scan report for 10.0.30.165
Host is up (0.094s latency).
Nmap scan report for 10.0.30.169
Host is up (0.052s latency).
Nmap scan report for 10.0.30.170
Host is up (0.0051s latency).
Nmap scan report for 10.0.30.172
Host is up (0.084s latency).
Nmap scan report for 10.0.30.181
Host is up (0.048s latency).
Nmap scan report for 10.0.30.190
Host is up (0.0048s latency).
Nmap scan report for 10.0.30.191
Host is up (0.032s latency).
Nmap scan report for 10.0.30.193
Host is up (0.016s latency).
Nmap scan report for 10.0.30.197
Host is up (0.0073s latency).
Nmap scan report for 10.0.30.200
Host is up (0.014s latency).
Nmap scan report for 10.0.30.209
Host is up (0.0089s latency).
Nmap scan report for 10.0.30.231
Host is up (0.041s latency).
Nmap scan report for 10.0.30.243
Host is up (0.0072s latency).
Nmap scan report for 10.0.30.252
Host is up (0.0071s latency).
Nmap done: 256 IP addresses (34 hosts up) scanned in 3.86 seconds
```

```
/nmap 10.0.30.1                                              <

Starting Nmap 6.49BETA4 ( https://nmap.org ) at 2016-01-15 11:43 CET
mass_dns: warning: Unable to open /etc/resolv.conf. Try using --system-dns or specify valid servers with --dns-servers
mass_dns: warning: Unable to determine any DNS servers. Reverse DNS is disabled. Try using --system-dns or specify valid servers with --dns-servers
Note: Host seems down. If it is really up, but blocking our ping probes, try -Pn
Nmap done: 1 IP address (0 hosts up) scanned in 1.01 seconds
******************************************************************************

/nmap 10.0.30.1 --unprivileged                                                <

Starting Nmap 6.49BETA4 ( https://nmap.org ) at 2016-01-13 09:41 CET
mass_dns: warning: Unable to open /etc/resolv.conf. Try using --system-dns or specify valid servers with --dns-servers
mass_dns: warning: Unable to determine any DNS servers. Reverse DNS is disabled. Try using --system-dns or specify valid servers with --dns-servers
Nmap scan report for 10.0.30.1
Host is up (0.0048s latency).
Not shown: 998 closed ports
PORT   STATE SERVICE
22/tcp open  ssh
23/tcp open  telnet

Nmap done: 1 IP address (1 host up) scanned in 1.76 seconds
```

**Comment by kost** [comment]
Thanks for reporting. Do you experience same issue with newer releases like:
https://github.com/kost/nmap-android/releases/tag/v7.12
or 
https://github.com/kost/nmap-android/releases/tag/v7.25BETA1

**Comment by FernandoDoming** [comment]
v7.12 does not work on my device as the provided ARM64 executable can not be linked. v7.25 works just fine.

```
# ./nmap scanme.nmap.org            

Starting Nmap 7.25BETA1 ( https://nmap.org ) at 2016-08-02 11:18 CEST
mass_dns: warning: Unable to open /etc/resolv.conf. Try using --system-dns or specify valid servers with --dns-servers
mass_dns: warning: Unable to determine any DNS servers. Reverse DNS is disabled. Try using --system-dns or specify valid servers with --dns-servers
Nmap scan report for scanme.nmap.org (45.33.32.156)
Host is up (0.23s latency).
Other addresses for scanme.nmap.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
Not shown: 996 closed ports
PORT      STATE SERVICE
22/tcp    open  ssh
80/tcp    open  http
9929/tcp  open  nping-echo
31337/tcp open  Elite

Nmap done: 1 IP address (1 host up) scanned in 3.06 seconds
```

```
# ./nmap scanme.nmap.org --unprivileged                                          <

Starting Nmap 7.25BETA1 ( https://nmap.org ) at 2016-08-02 11:18 CEST
mass_dns: warning: Unable to open /etc/resolv.conf. Try using --system-dns or specify valid servers with --dns-servers
mass_dns: warning: Unable to determine any DNS servers. Reverse DNS is disabled. Try using --system-dns or specify valid servers with --dns-servers
Nmap scan report for scanme.nmap.org (45.33.32.156)
Host is up (0.20s latency).
Other addresses for scanme.nmap.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
Not shown: 996 closed ports
PORT      STATE SERVICE
22/tcp    open  ssh
80/tcp    open  http
9929/tcp  open  nping-echo
31337/tcp open  Elite

Nmap done: 1 IP address (1 host up) scanned in 14.83 seconds
```

**Comment by kost** [comment]
Thanks for extensive reporting. I have released 7.31 version, you just need to download with default update URL and everything should work now.
